suspendmanager.lua

```diff
@@ -16,23 +16,41 @@ end
 local GLOBAL_KEY = 'suspendmanager' -- used for state change hooks and persistence
 
 enabled = enabled or false
-preventblocking = preventblocking == nil and true or preventblocking
 
 eventful.enableEvent(eventful.eventType.JOB_INITIATED, 10)
 eventful.enableEvent(eventful.eventType.JOB_COMPLETED, 10)
 
+SuspendManager = defclass(SuspendManager)
+SuspendManager.ATTRS {
+    --- When enabled, suspendmanager also tries to suspend blocking jobs,
+    --- when not enabled, it only cares about avoiding unsuspending jobs suspended externally
+    preventBlocking = false,
+    --- Jobs that are on a tile with a designation (smooth, carve track, engrave)
+    jobsOnDesignation = {},
+}
+
+function SuspendManager.new(preventBlocking)
+    local manager = SuspendManager {
+        preventBlocking = preventBlocking
+    }
+    return manager
+end
```

Comment:
if there's no singleton, what is the advantage of hiding instantiation behind a method? Normally, clients would call SuspendManager{} directly.

Reply:
Yep, I'm just so used to languages with constructors that I did the same here - removed